src/test/java/org/prebid/server/bidder/emx_digital/EmxDigitalBidderTest.java

```diff
@@ -0,0 +1,415 @@
+				.build()))
+			.id("request_id")
+			.build();
+
+		// when
+		final Result<List<HttpRequest<BidRequest>>> result = emxDigitalBidder
+			.makeHttpRequests(bidRequest);
+
+		// then
+		assertThat(result.getErrors()).hasSize(2);
+		assertThat(result.getErrors().get(0).getMessage())
+			.startsWith("tagid cant be 0");
+		assertThat(result.getErrors().get(1).getMessage())
+			.startsWith("Error in makeBidRequest of Imp");
+		assertThat(result.getValue()).isEmpty();
+	}
+
+	@Test
+	public void makeHttpRequestsShouldReturnErrorWhenWAndHIsNullAndBannerFormatIsNull() {
+		// given
+		final BidRequest bidRequest = BidRequest.builder()
+			.imp(singletonList(Imp.builder()
+				.banner(Banner.builder().build())
+				.ext(mapper.valueToTree(ExtPrebid.of(null, ExtImpEmxDigital.of("1", "1"))))
+				.build()))
+			.id("request_id")
+			.build();
+
+		// when
+		final Result<List<HttpRequest<BidRequest>>> result = emxDigitalBidder
+			.makeHttpRequests(bidRequest);
+
+		// then
+		assertThat(result.getErrors()).hasSize(2);
+		assertThat(result.getErrors().get(0).getMessage())
+			.startsWith("Need at least one size to build request");
+		assertThat(result.getErrors().get(1).getMessage())
+			.startsWith("Error in makeBidRequest of Imp");
+		assertThat(result.getValue()).isEmpty();
+	}
+
+	@Test
+	public void makeHttpRequestsShouldReturnErrorWhenWAndHIsNullAndBannerFormatIsEmpty() {
+		// given
+		final BidRequest bidRequest = BidRequest.builder()
+			.imp(singletonList(Imp.builder()
+				.banner(Banner.builder().format(Collections.emptyList()).build())
+				.ext(mapper.valueToTree(ExtPrebid.of(null, ExtImpEmxDigital.of("1", "1"))))
+				.build()))
+			.id("request_id")
+			.build();
+
+		// when
+		final Result<List<HttpRequest<BidRequest>>> result = emxDigitalBidder
+			.makeHttpRequests(bidRequest);
+
+		// then
+		assertThat(result.getErrors()).hasSize(2);
+		assertThat(result.getErrors().get(0).getMessage())
+			.startsWith("Need at least one size to build request");
+		assertThat(result.getErrors().get(1).getMessage())
+			.startsWith("Error in makeBidRequest of Imp");
+		assertThat(result.getValue()).isEmpty();
+	}
+
+	@Test
+	public void makeHttpRequestsShouldModifyImpWhenExtImpEmxDigitalContainsRequredValues() {
+		// given
+		final BidRequest bidRequest = BidRequest.builder()
+			.imp(singletonList(Imp.builder()
+				.banner(Banner.builder().w(100).h(100).build())
+				.ext(mapper.valueToTree(ExtPrebid.of(null, ExtImpEmxDigital.of("123", "2"))))
+				.build()))
+			.site(Site.builder().page("https://exmaple/").build())
+			.id("request_id")
+			.build();
+
+		// when
+		final Result<List<HttpRequest<BidRequest>>> result = emxDigitalBidder
+			.makeHttpRequests(bidRequest);
+
+		// then
+		assertThat(result.getErrors()).isEmpty();
+		assertThat(result.getValue()).hasSize(1)
+			.extracting(httpRequest -> mapper.readValue(httpRequest.getBody(), BidRequest.class))
+			.flatExtracting(BidRequest::getImp)
+			.hasSize(1)
+			.first()
```

Comment:
You can simplify it by using instead a `.containsOnly(Imp.. imp)` verification and passing an Imp object with expected fields. I mean this one is cool from testing viewpoint, but looks too robust. Thanks!